tests/test_warnings.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+
+from pystac_client import Client
+from pystac_client.warnings import DoesNotConformTo, ignore, strict
+
+from .helpers import TEST_DATA
+
+
+class TestWarningContextManagers:
+    @pytest.mark.filterwarnings("error")
```

Comment:
You _probably_ don't need to explicitly include these since we run the tests with the `-Werror` flag on CI. It could be a nice improvement to add `-Werror` to the pytest `addopts` in pyproject.toml actually...